### PR TITLE
final commit

### PR DESCRIPTION
**Pull request by TanerIlyazov**

### CODE REVIEW

File: src/com/teodora/Author.java [hunk 1]
```diff
@@ -2,21 +2,25 @@
 
 public class Author {
 
-    String name="andi";
-    int age=30;
+    String name = "andi";
```
Comment (TanerIlyazov) [inline src/com/teodora/Author.java:5]: In this way you can only have 1 author. You need to initialize these properties of an Author in a constructor, so you can create several authors.

File: src/com/teodora/Author.java [hunk 2]
```diff
@@ -2,21 +2,25 @@
...
 
-        System.out.println("Author Name:"+name + " "+"and age: "+age);
+        System.out.println("Author Name:" + name + " " + "and age: " + age);
     }
-/**second method witch is going to output authors formatted song and name */
+
+    /**
+     * second method witch is going to output authors formatted song and name
+     */
     public void formattedAge() {
```
Comment (TanerIlyazov) [inline src/com/teodora/Author.java:22]: Method is called formatted age, but you're not outputting age.

File: src/com/teodora/Author.java [hunk 3]
```diff
@@ -2,21 +2,25 @@
...
- * to output authors formatted name and age
- */
+    /**
+     * Creating the first method witch is going
+     * to output authors formatted name and age
+     */
     public void formattedNameAuthor() {
 
-        System.out.println("Author Name:"+name + " "+"and age: "+age);
+        System.out.println("Author Name:" + name + " " + "and age: " + age);
```
Comment (TanerIlyazov) [inline src/com/teodora/Author.java:16]: You should use println in the Author class. You should have a method, which returns the String which will represent the Author's info. Check Object.toString() method.

File: src/com/teodora/Author.java [hunk 4]
```diff
@@ -2,21 +2,25 @@
 
 public class Author {
 
-    String name="andi";
-    int age=30;
+    String name = "andi";
+    int age = 30;
     String song = "la la la ";
```
Comment (TanerIlyazov) [inline src/com/teodora/Author.java:7]: You have a Song.class, but you're using string here?

File: src/com/teodora/Song.java [hunk 5]
```diff
@@ -7,31 +7,42 @@
     String timing = "3:00";
     String lyrics = "hello hello hello";
 
-    /**Creating the first method witch is going
-     *  to output the characteristics of the class */
+    /**
+     * Creating the first method witch is going
```
Comment (TanerIlyazov) [inline src/com/teodora/Song.java:11]: Calling this the first method isn't a good practice, you could change the order of methods in the future.

File: src/com/teodora/Song.java [hunk 6]
```diff
@@ -7,31 +7,42 @@
...
+     * to output the characteristics of the class
+     */
 
     public void printCharacteristics() {
 
         System.out.println("Song title:" + " " + title + " " + "Song genre:" + " "
                 + genre + " " + "Song timing:" + " " + timing + " " + "Song lyric:"
```
Comment (TanerIlyazov) [inline src/com/teodora/Song.java:18]: "Song timing:" + " " -> "Song timing: "

File: src/com/teodora/Song.java [hunk 7]
```diff
@@ -7,31 +7,42 @@
...
                 + " " + lyrics);
     }
-    /** method witch is going to check if the genre is correct*/
+
+    /**
+     * method witch is going to check if the genre is correct
+     */
 
     public void songGenre(String genr) {
         if (genr == genre) {
```
Comment (TanerIlyazov) [inline src/com/teodora/Song.java:27]: genr?

Comment (TanerIlyazov) [reply src/com/teodora/Song.java:27]: see string comparison methods

File: src/com/teodora/Song.java [hunk 8]
```diff
@@ -7,31 +7,42 @@
...
         } else {
-            System.out.println("Wrong, it is not"+genr);
+            System.out.println("Wrong, it is not" + genr);
         }
     }
-    /** method witch is going to output song timing*/
+
+    /**
+     * method witch is going to output song timing
+     */
```
Comment (TanerIlyazov) [inline src/com/teodora/Song.java:37]: Why is there a blank line in-between

File: src/com/teodora/Song.java [hunk 9]
```diff
@@ -7,31 +7,42 @@
...
 
     public void songTiming() {
```
Comment (TanerIlyazov) [inline src/com/teodora/Song.java:39]: blank line

File: src/com/teodora/Song.java [hunk 10]
```diff
@@ -7,31 +7,42 @@
     String timing = "3:00";
```
Comment (TanerIlyazov) [inline src/com/teodora/Song.java:7]: This class is not reusable, hence breaks all OOP conventions. :)

File: src/com/teodora/Author.java [hunk 11]
```diff
@@ -2,25 +2,35 @@
 
 public class Author {
 
-    String name = "andi";
-    int age = 30;
+    String name;
+    int age;
     String song = "la la la ";
 
+    /*** creating a constructor to initialize the properties of an Author
```
Comment (TanerIlyazov) [inline src/com/teodora/Author.java:9]: The 3 * stars things was miss-poken today. It's just /**

File: src/com/teodora/Author.java [hunk 12]
```diff
@@ -2,25 +2,35 @@
...
+     */
+    public Author(String name, int age) {
```
Comment (TanerIlyazov) [inline src/com/teodora/Author.java:11]: what if the client does new Author(null, -1)

File: src/com/teodora/Author.java [hunk 13]
```diff
@@ -2,25 +2,35 @@
...
+    /*** Creating method witch is going to output authors formatted name and age
      */
-    public void formattedNameAuthor() {
+    public void formattedNameAndAge() {
+
+        String formattedName = name.substring(0, 1).toUpperCase() + name.substring(1);
 
-        System.out.println("Author Name:" + name + " " + "and age: " + age);
+        int formattedAge = Math.round(age);
+        System.out.println("Author name: " + formattedName);
```
Comment (TanerIlyazov) [inline src/com/teodora/Author.java:23]: Don't use system out print inside Domain object classes. f.e. You can use it in the main method of Application.class, but not in Author or Song.

File: src/com/teodora/Author.java (hunk 4 above)
Comment (TanerIlyazov) [reply src/com/teodora/Author.java:7]: Still not using Song class inside Author

File: src/com/teodora/Author.java [hunk 14]
```diff
@@ -2,25 +2,35 @@
...
 
-    /**
-     * Creating the first method witch is going
-     * to output authors formatted name and age
+    /*** Creating method witch is going to output authors formatted name and age
      */
-    public void formattedNameAuthor() {
+    public void formattedNameAndAge() {
+
+        String formattedName = name.substring(0, 1).toUpperCase() + name.substring(1);
```
Comment (TanerIlyazov) [inline src/com/teodora/Author.java:20]: It's the responsibility of the person who is entering his name to make sure it start with Uppercase. You shouldn't change his name while outputting it.

File: src/com/teodora/Author.java [hunk 15]
```diff
@@ -2,25 +2,35 @@
...
 
-        System.out.println("Author Name:" + name + " " + "and age: " + age);
+        int formattedAge = Math.round(age);
```
Comment (TanerIlyazov) [inline src/com/teodora/Author.java:22]: why?

File: src/com/teodora/Author.java [hunk 16]
```diff
@@ -2,25 +2,35 @@
...
     }
 
-    /**
-     * second method witch is going to output authors formatted song and name
-     */
-    public void formattedAge() {
 
-        System.out.println("Authors song:" + " " + song + " " + "Authors name: " + name);
+    /*** method witch is going to output authors formatted song and name*/
+    public void formattedTitleAndAuthor() {
```
Comment (TanerIlyazov) [inline src/com/teodora/Author.java:29]: Same remarks as the upper ones

File: src/com/teodora/Song.java [hunk 17]
```diff
@@ -2,39 +2,37 @@
 
 public class Song {
```
Comment (TanerIlyazov) [inline src/com/teodora/Song.java:3]: This class could use some explanation

File: src/com/teodora/Song.java [hunk 18]
```diff
@@ -2,39 +2,37 @@
...
-    String genre = "pop";
-    String timing = "3:00";
-    String lyrics = "hello hello hello";
-
-    /**
-     * Creating the first method witch is going
-     * to output the characteristics of the class
-     */
+    String title;
+    String genre;
```
Comment (TanerIlyazov) [inline src/com/teodora/Song.java:6]: Can you have a genre like "asd"?

File: src/com/teodora/Song.java [hunk 19]
```diff
@@ -2,39 +2,37 @@
...
+    double timing;
+    String lyrics;
+
+    public Song(String title,String genre,double timing,String lyrics) {
+        this.title=title;
```
Comment (TanerIlyazov) [inline src/com/teodora/Song.java:11]: formatting

File: src/com/teodora/Song.java [hunk 20]
```diff
@@ -2,39 +2,37 @@
...
+        this.genre=genre;
+        this.timing=timing;
+        this.lyrics=lyrics;
+    }
 
+        /*** method witch is going to output the characteristics of the class*/
```
Comment (TanerIlyazov) [inline src/com/teodora/Song.java:17]: formatting and /**

File: src/com/teodora/Song.java [hunk 21]
```diff
@@ -2,39 +2,37 @@
...
     public void printCharacteristics() {
 
-        System.out.println("Song title:" + " " + title + " " + "Song genre:" + " "
-                + genre + " " + "Song timing:" + " " + timing + " " + "Song lyric:"
-                + " " + lyrics);
+        System.out.println("Song title: " + title);
```
Comment (TanerIlyazov) [inline src/com/teodora/Song.java:20]: check toString()

File: src/com/teodora/Song.java [hunk 22]
```diff
@@ -2,39 +2,37 @@
...
      * method witch is going to check if the genre is correct
      */
-
-    public void songGenre(String genr) {
-        if (genr == genre) {
-            System.out.println("Correct" + " " + genre);
-        } else {
-            System.out.println("Wrong, it is not" + genr);
-        }
+    public void songGenre(String genre) {
```
Comment (TanerIlyazov) [inline src/com/teodora/Song.java:29]: method name does not match the implementation

File: src/com/teodora/Author.java (hunk 4 above)
Comment (vanyamihova-deltasource) [reply src/com/teodora/Author.java:7]: Access modifiers!

File: src/com/teodora/Song.java [hunk 23]
```diff
@@ -2,39 +2,44 @@
...
-    String genre = "pop";
-    String timing = "3:00";
-    String lyrics = "hello hello hello";
-
-    /**Creating the first method witch is going
-     *  to output the characteristics of the class */
-
-    public void printCharacteristics() {
+    String title;
+    String genre;
```
Comment (AleksanderDzhongarov) [inline src/com/teodora/Song.java:6]: put some access modifiers

File: src/com/teodora/Song.java [hunk 24]
```diff
@@ -2,39 +2,44 @@
...
+    double timing;
+    String lyrics;
+
+    public Song(String title, String genre, double timing, String lyrics) {
+        this.title = title;
```
Comment (AleksanderDzhongarov) [inline src/com/teodora/Song.java:11]: add custom validation

**Pull request merged**